### PR TITLE
Allow FTPHook to change port number

### LINKED ISSUE #39308: FTPHook doesn't not allow to change port
### Apache Airflow Provider(s)

ftp

### Versions of Apache Airflow Providers

3.8.0

### Apache Airflow version

2.6.3

### Operating System

Ubuntu 20.04

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Irrelevant as it is in source code

### What happened

When setting up an FTP connection with a custom port (different from 21), the FTP Hook does not read it at all and simply passes the hostname and uses the default port set by FTP_PORT in ftplib.FTP_PORT (used by ftplib.FTP.port), thus preventing any modification of default port 21. 
PS: The 'connect' function of the FTP class reads port from the self.port attribute;

### What you think should happen instead

The FTP Hook should be able to read the port (if set) and pass it to the FTP object created 

### How to reproduce

Set up any FTP connection with and test it on Airflow UI. 

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by VelmiraPetkova**
Allow FTPHook to use a custom FTP port number

closes: #39308

### CODE REVIEW

**Review by dirrao — CHANGES_REQUESTED** [review]
Congratulations on your first PR. 
Can you add the test case for custom port?

File: airflow/providers/ftp/hooks/ftp.py [hunk 1]
```diff
@@ -58,7 +58,11 @@ def get_conn(self) -> ftplib.FTP:
         if self.conn is None:
             params = self.get_connection(self.ftp_conn_id)
             pasv = params.extra_dejson.get("passive", True)
-            self.conn = ftplib.FTP(params.host, params.login, params.password)  # nosec: B321
+            self.conn = ftplib.FTP()  # nosec: B321
+            if params.host:
+                self.conn.connect(params.host, params.port)
+                if params.login:
+                    self.conn.login(params.login, params.password)
```
Comment (vincbeck) [inline airflow/providers/ftp/hooks/ftp.py:72]: Can `params.login` be provided but not `params.host`? If yes, then there is an indentention issue

Comment (VelmiraPetkova) [reply airflow/providers/ftp/hooks/ftp.py:72]: This is implemented in the same way as in the standard library. Plus, `params.login` and `params.password` only have a meaning when trying to connect to a FTP server for which you need to have a host name supplied (even if it's localhost).

**Comment by VelmiraPetkova** [comment]
I did some tests and fixed a tiny bug, when no port is supplied.

File: airflow/providers/ftp/hooks/ftp.py (hunk 1 above)
Comment (vincbeck) [reply airflow/providers/ftp/hooks/ftp.py:72]: No problem then

**Review by vincbeck — APPROVED** [review]

File: airflow/providers/ftp/hooks/ftp.py [hunk 2]
```diff
@@ -58,7 +58,14 @@ def get_conn(self) -> ftplib.FTP:
         if self.conn is None:
             params = self.get_connection(self.ftp_conn_id)
             pasv = params.extra_dejson.get("passive", True)
-            self.conn = ftplib.FTP(params.host, params.login, params.password)  # nosec: B321
+            self.conn = ftplib.FTP()  # nosec: B321
+            if params.host:
+                port = ftplib.FTP_PORT
+                if params.port is not None:
+                    port = params.port
```
Comment (amoghrajesh) [inline airflow/providers/ftp/hooks/ftp.py:68]: Can we add a log statement here so that we know which port was used

Comment (VelmiraPetkova) [reply airflow/providers/ftp/hooks/ftp.py:68]: Hello, I added a log message. Can you take a look?

**Review by amoghrajesh — APPROVED** [review]
Thanks @VelmiraPetkova
LGTM +1

**Comment by eladkal** [comment]
@VelmiraPetkova can you please fix static checks?

**Comment by VelmiraPetkova** [comment]
Sorry for the delay. I updated the PR. Could you please take a look? Thank you!

**Review by amoghrajesh — APPROVED** [review]
LGTM. Thanks @VelmiraPetkova!

@dirrao wdyt?

File: airflow/providers/ftp/hooks/ftp.py [hunk 3]
```diff
@@ -19,10 +19,12 @@
 
 import datetime
 import ftplib  # nosec: B402
+import logging
 from typing import Any, Callable
 
 from airflow.hooks.base import BaseHook
 
+logger = logging.getLogger(__name__)
```
Comment (eladkal) [inline airflow/providers/ftp/hooks/ftp.py]:
```suggestion
logger = logging.getLogger(__name__)


```

Fix static checks

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
The failing integration test case is a flake - mssql does not have enough time to start. This has been fixed in #39866

**Pull request merged**